runtime/compiler/control/CompilationThread.cpp

```diff
@@ -9248,6 +9248,31 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
                reloRuntime,
                &target);
 
+#if defined(J9VM_OPT_JITSERVER)
+if (that->_methodBeingCompiled->isOutOfProcessCompReq())
+   {
+   // use the default code cache
+   compiler->getOptions()->setCodeCacheKind(TR::CodeCacheKind::DEFAULT_CC);
+   }
+else
+#endif
+   {
+   TR::SimpleRegex *regex = compiler->getOptions()->getTransientClassRegex();
+   if (regex)
+      {
+      J9Method *method = that->_methodBeingCompiled->getMethodDetails().getMethod();
```

Comment:
I see that "methodDetails" are cached somewhere above, so you can simplify this line to just
`J9Method *method = details.getMethod();`